packages/adapters-library/src/defiProvider.ts

```diff
@@ -429,13 +445,20 @@ export class DefiProvider {
       return { tokens }
     }
 
-    return this.runForAllProtocolsAndChains({
-      runner,
-      filterProtocolIds,
-      filterChainIds,
+    const result = (
+      await this.runForAllProtocolsAndChains({
+        runner,
+        filterProtocolIds,
+        filterChainIds,
 
-      method: 'unwrap',
-    })
+        method: 'unwrap',
+      })
+    ).filter(
+      (result) =>
+        !result.success || (result.success && result.tokens.length > 0),
+    )
+
```

Comment:
Filter out empty responses this can happen when a filterProtocolToken filter is included where the protocol has many adapters